### ISSUE TITLE
[BUG] MSVC 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease'

### ISSUE THREAD

**Issue by lishaojiang** [google/benchmark#1655]
**Describe the bug**

I have download and compile and test Google BenchMark success on My Linux, My Mac.  but On My Windows PC , I get an error when other project link the benchmark!

I know this is because the benchmark project is a lib generated by **MD**, but the test project regards it as **MT**, so the link is wrong. 

I am forced to manually modify the benchmark attribute to **MT** then build  and then manually copy it to the installation directory . it can work well.

I have tried the code below adding to CMakeLists.txt, it don't work!
```
if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
    add_compile_options(/MT)
endif()
```
or adding sa to the build script, but it has no effect!
`cmake -E chdir "build" cmake -DCMAKE_CXX_FLAGS_RELEASE=/MT -DCMAKE_CXX_FLAGS=/MT -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DCMAKE_BUILD_TYPE=Release ../`

**System**
Which OS, compiler, and compiler version are you using:
  - OS: windows 11,22H2,22621.2134
  - Compiler and version: Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.6.5

**To reproduce**
install.bat
```
git clone https://github.com/google/benchmark.git
cd benchmark
cmake -E make_directory "build"
cmake -E chdir "build" cmake -DCMAKE_CXX_FLAGS_RELEASE=/MT -DCMAKE_CXX_FLAGS=/MT -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DCMAKE_BUILD_TYPE=Release ../
cmake --build "build" --config Release --target install
```
then other project use the google bench

CMakeLists.txt
```
find_package(benchmark REQUIRED)
target_link_libraries(main PUBLIC benchmark::benchmark)
target_link_libraries(main PUBLIC benchmark::benchmark_main) 
```

**Expected behavior**
How Can fix the google benchmark MD to MT auto

**Screenshots**
![image](https://github.com/google/benchmark/assets/23740323/8ecc79fb-2501-47e9-b0b1-d94d801d3e63)

**Additional context**

**Comment by lishaojiang** [comment]
well done , I find one solution！
Just use bleow code in Google BenchMark every CMakeLists.txt file！ I find 4 cmakelists.txt file in my pc!
```
if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
    add_compile_options(/MT)
endif()
```

The original method didn't work, anyone know why
```
if (MSVC)
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MT")
endif
```

**Comment by imrichardcole** [comment]
@dmah42 protobuf allows this build option:

`protobuf_MSVC_STATIC_RUNTIME=ON/OFF`

Do we want a similar setting in benchmark?

**Comment by dmah42** [comment]
we have BUILD_SHARED_LIBS but that seems to only be changing how we export symbols.  i was sure we had a flag in the build to switch between shared and static libs.

**Comment by imrichardcole** [comment]
Taking the example from protobuf, setting `-Dprotobuf_MSVC_STATIC_RUNTIME=ON` produces an /MT(d) build.  Not specifying (or setting to OFF) defaults to /MD(d):

![image](https://github.com/user-attachments/assets/040ab527-3e69-4a0f-8162-a2c149d8d68f)

If you agree you'd like the equivalent to be done in benchmark i.e `-Dbenchmark_MSVC_STATIC_RUNTIME=ON` I'll get it done.

**Comment by dmah42** [comment]
yes, but i would hope we don't need an MSVC specific flag.  we should have a flag that switches MT to MD on msvc and linux, presumably by adding the `STATIC` flag to the `add_library` call in cmake.

**Comment by imrichardcole** [comment]
Noted, shall share a branch once I have something to show.
